dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/protocol/ProtocolFilterWrapper.java

```diff
@@ -169,8 +169,6 @@ public Result invoke(Invocation invocation) throws RpcException {
                                 listener.onError(t, filterInvoker, invocation);
                             }
                         }
-                    } else {

```

Comment:
would you mind to explain why it is unnecessary?

Reply:
We cannot remove this method, otherwise, legacy filters will not get invoked.